master/internal/api_experiment.go

```diff
@@ -1922,6 +1934,477 @@ func sortExperiments(sortString *string, experimentQuery *bun.SelectQuery) error
 	return nil
 }
 
+func getFilterMetricPrefixes() map[string]string {
+	// Prefixes for column names matching to metric related tables
+	metricPrefixes := map[string]string{
+		"validation.": "e.validation_metrics->>'%s'",
+		"hp.":         "e.config->'hyperparameters'->'%s'->>'val'",
```

Comment:
does this only work for constant hyperparameters? should we support other types as well? https://docs.determined.ai/latest/reference/reference-training/experiment-config-reference.html#hyperparameters

Reply:
I tried to parse the query in the most flexible way possible, I think most of the PR can be unchanged in order to support categorical or other types of `hyperparameters`. What is missing here is if we want to be able to perform some sort of `contains` for a `hyperparameter` that could be a list, that  would need new parsing logic. I think we don't have this supported on the front end yet however. The operators we have are >,<,~,: to signify greater or less than, like, or equal to. 

If we are going to support `contains` for a hyperparameter which has a list  for a value we might need to think about how that will work in practice. For example will we say `hp.hyperparameter~5` should perform a `contains` on a list if that hyperparameter is a list? It may be easier to create a new symbol for the query language specifically for this use case.

A reason that this is not as clear is because we did base the language off of the stripe query language which does not support `list` based queries it seems: https://stripe.com/docs/search#search-syntax

cc: @keita-determined @NicholasBlaskey since this will need to be addressed.

Reply:
I think we went over this in sprint planning -- we do expect categorical hyperparameters to behave like text w/r/t filtering in that it supports equality or contains (via the like operator), but not greater than or less than.
Absent that, we still should support int, double and log hyperparameters, the config of which does not use `val` in the config.